bin/version-check.js

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env node
+/** @format */
+/* eslint-disable no-console, no-process-exit */
+
+/**
+ * Checks the user's node and npm versions to make sure the major
+ * versions match. This script will exit with a non-zero status
+ * (causing the build to fail) if the major version of `node` and
+ * `npm` don't match.
+ */
+
+const chalk = require( 'chalk' );
+const checkNodeVersion = require( 'check-node-version' );
+const semver = require( 'semver' );
+const packageJSON = require( '../package.json' );
+
+const majorNodeVersion = semver.major( packageJSON.engines.node );
+const majorNpmVersion = semver.major( packageJSON.engines.npm );
+
+function buildMessage( { engine, engines, results } ) {
+	const icon = results[ engine ].isSatisfied ? chalk.green( '✓' ) : chalk.red( '𝘅' );
+	let message = `${ icon } expecting ${ engine }: ${ chalk.blue( engines[ engine ] ) }`;
+
+	if ( results[ engine ].isSatisfied ) {
+		message += ` (Your version: ${ chalk.green( results[ engine ].version.version ) })`;
+	} else {
+		// eslint-disable-next-line max-len
+		message += ` (Your version: ${ chalk.red( results[ engine ].version.version ) } is not in valid range: ${ chalk.blue( results[ engine ].wanted.range ) })`;
+	}
```

Comment:
there's no real need for mutation here. we can just as easily create a separate second string and return the combined forms

```js
const expected = `${ icon } expecting ${ engine }: …`;
const actual = results[ engine ].isSatisfied
	? ` (Your version: ${ ch… })`
	: ` (Your version: ${ ch…})`;

return expected + actual;
```

Reply:
Oh, that's true. I can send a quick follow-up patch in a bit. The original message structure lent itself more to mutation and I think this is just a remnant of that. 👍